tests/numerics/nonlinear/test_nonlinear_solvers.py

```diff
@@ -0,0 +1,61 @@
+import porepy as pp
+import numpy as np
+from typing import Any
+
+from porepy.models.fluid_mass_balance import SinglePhaseFlow
+
+
+class NonlinearSinglePhaseFlow(SinglePhaseFlow):
+    """Class setup which forces a set number of nonlinear iterations to be run for
+    testing."""
+
+    def _is_nonlinear_problem(self):
+        """Method for stating that the problem is nonlinear."""
+        return True
+
+    def check_convergence(
+        self,
+        nonlinear_increment: np.ndarray,
+        residual: np.ndarray,
+        reference_residual: np.ndarray,
+        nl_params: dict[str, Any],
+    ) -> tuple[bool, bool]:
+        """Method for checking convergence.
+
+        This method is only used for testing. It returns not converged if the iteration
+        count is smaller than a pre set value.
+
+        """
+        diverged = False
+        if (
+            self.nonlinear_solver_statistics.num_iteration
+            < self.expected_number_of_iterations
+        ):
+            converged = False
+            return converged, diverged
+        converged = True
+        return converged, diverged
+
+    def after_nonlinear_failure(self) -> None:
+        """Method which is called if the non-linear solver fails to converge."""
+        self.save_data_time_step()
```

Comment:
Why is this needed?

Reply:
Hm, apparently it is not needed. I had some error messages when I made test, and including this method (seemingly) removed the problem. I see now that it is not needed, so I will delete it.